client.go

```diff
@@ -728,6 +741,11 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		// without racing against the closeBody call in persistConn.writeLoop.
 		httpreq := *req.Request
 		req.Request = &httpreq
+
+		if err := c.Postprocess(req.Request); err != nil {
```

Comment:
Since the Client is exported, the creation may not be done by `NewClient` but simply by creating a struct and passing it into further processing. In such case the Postprocess would be nil. Would it break processing?

Reply:
The same issue will be with existing `CheckRetry` or `Backoff`